map/src/main/kotlin/edu/artic/map/MapObjectDetailsViewModel.kt

```diff
@@ -92,8 +136,10 @@ class MapObjectDetailsViewModel @Inject constructor(val analyticsTracker: Analyt
 
 
     fun playCurrentObject() {
-        articObject?.let {
-            playerControl.onNext(PlayerAction.Play(it))
+        articObject?.let { source: ArticObject ->
```

Comment:
this is where `Observables.combineLatest()` would come in handy using the `objectObservable` objects.